### PR TITLE
Fix pricing info and exclude eks nodegroup instances

### PR DESCRIPTION
**Pull request by CodyRay**
(this PR is best reviewed commit by commit)

This PR:

* Runs all the IntelliJ inspections and squashes all the warnings and formatting issues in the nagbot codebase
* Adds a dryrun option
* Upgrades the 4 dependencies
* Replaces the awspricing dependency with direct calls to the pricing api
* Handle the case of an invalid date by ignoring the date
* Ignore EKS nodegroup instances (IMO they were causing too much noise)
* Takes some first steps to Dockerize NagBot

The awspricing dependency uses a tremendously inefficient method of querying prices. It loads all the prices for everything in EC2 and then we query the gigantic corpus of prices that was pulled down. I replaced that with direct queries to the pricing api for one request for each instance type, operating system, and region combo.

image

### CODE REVIEW

File: docker-run.sh [hunk 1]
```diff
@@ -0,0 +1,17 @@
+#/bin/bash
+
+docker run \
+  -e "SLACK_BOT_TOKEN=" \
+  -e "AWS_ACCESS_KEY_ID=" \
+  -e "AWS_SECRET_ACCESS_KEY=" \
+  -e "GDOCS_SERVICE_ACCOUNT_FILENAME=/gdocs-service-account.json" \
```
Comment (spaceval) [inline docker-run.sh:7]: We should probably refactor this to post results elsewhere, but it's probably out of scope for this ticket. I'm not sure who owns the Google account this currently goes to, but I'd assume it's Stephen's.

**Review by spaceval — APPROVED** [review]
All of this looks amazing! So many improvements. And I can't thank you enough for setting aside the bandwidth to get this done.

Any idea if it'd be a big pull to also tack this on? 😛 https://seeq.atlassian.net/browse/CRAB-22304

File: docker-run.sh (hunk 1 above)
Comment (CodyRay) [reply docker-run.sh:7]: Stephen transferred the ownership of the google doc itself; I don't have any idea how this service account works though or how it is created. But I agree, anything Seeq + Google Account is really sketchy though.

**Comment by CodyRay** [comment]
I looked at CRAB-22304 earlier, but couldn't figure it out, but I took another look and it was indeed an easy fix

**Comment by spaceval** [comment]
Oh, you are just awesome! 🤩

**Pull request merged**